File: src/main/java/net/yenil/itemunbreakable/utility/ItemUtils.java
package net.yenil.itemunbreakable.utility;

import org.bukkit.inventory.ItemStack;
import org.bukkit.inventory.meta.ItemMeta;

import java.lang.reflect.Method;

public class ItemUtils {

    public static ItemStack setUnbreakable(ItemStack itemstack, boolean unbreakable) {
        ItemMeta meta = itemstack.getItemMeta();
        try {
            Method metaSpigot = ItemMeta.class.getDeclaredMethod("spigot");
            Method setUnbreakable = metaSpigot.getReturnType().getDeclaredMethod("setUnbreakable", boolean.class);
            Object spigot = metaSpigot.invoke(meta);
            setUnbreakable.invoke(spigot, unbreakable);
        } catch (Exception ex) {
            meta.setUnbreakable(unbreakable);
        }
        itemstack.setItemMeta(meta);
        return itemstack;
    }
}

File: src/main/java/net/choco/itemunbreakable/utility/ItemUtils.kt
package net.choco.itemunbreakable.utility

import org.bukkit.inventory.ItemStack
import org.bukkit.inventory.meta.ItemMeta

object ItemUtils {
    fun setUnbreakable(itemstack: ItemStack, unbreakable: Boolean): ItemStack {
        val meta = itemstack.itemMeta
        try {
            val metaSpigot = ItemMeta::class.java.getDeclaredMethod("spigot")
            val setUnbreakable =
                metaSpigot.returnType.getDeclaredMethod("setUnbreakable", Boolean::class.javaPrimitiveType)
            val spigot = metaSpigot.invoke(meta)
            setUnbreakable.invoke(spigot, unbreakable)
        } catch (ex: Exception) {
            meta!!.isUnbreakable = unbreakable
        }
        itemstack.itemMeta = meta
        return itemstack
    }
}